### DISCUSSION TITLE
How to update the parent's (stacklayout) autosizing? (Solved)

### DISCUSSION THREAD

**Discussion by WileyNg** [picoe/Eto#2178] [General]
![image](https://user-images.githubusercontent.com/5993201/161497433-6e7d9e2c-82bc-40b2-83fc-364bb69eb9d3.png)

![image](https://user-images.githubusercontent.com/5993201/161495829-0075f256-edf3-421d-92ac-1c2f1ed7d0f6.png)
![image](https://user-images.githubusercontent.com/5993201/161495868-080b3eea-f2fa-423e-9e54-fe4bfc71a341.png)

I made a custom-made expander box but i am getting a hard time to update the parent's autosizing, after the expander is expanded.
Is it possible to shed some light for me on this?

```C#
        public void OnExpanded()
        {
            Content = expanded ? Control : new DynamicLayout();
            
            int _ContentSize = (int)Content.GetPreferredSize().Height + (Padding * 4) + (int)textSize.Height;
            Height = _ContentSize;
            DrawGraphics();
            Parent.Parent.Invalidate(); ;
            Invalidate();
        }
```

I made a method which is called when the Header is clicked(MouseUp). 
The collapsed expander will just have an empty DynamicLayout, while the expanded one will have all the controls.
The height I get it with a GetPreferredSize. 

The problem is, the control (StackLayout) which the Custom-Expander is nested in, doesn't adjust the sizing itself whenever the child control height is changed.

I tried to call Invalidate() function for the StackLayout by finding the Parents of the custom-expander, but in vain.

**Comment by WileyNg** [comment]
Problem Solved

**Reply by cwensley** [reply]
Hey @WileyNg,

Thanks for letting us know you figured out the issue.  For future users it would be helpful to know what solved it.  

Also, there's no need to create an issue _and_ a discussion for next time.  We can always turn a discussion into an issue if it is found to be something that can be improved or changed within Eto.

Cheers!
